Resolve intersphinx to astropy as current build (#11690)
astropy/astropy: docs/changes/coordinates/11690.other.rst

> SkyCoord links now use intersphinx links to to work in affiliate packages.

Reviewer: Actually, maybe do expand that it now works if `SkyCoord` is subclassed (which I think is problematic for other reasons...)

Author: Done!

"SkyCoord links now use intersphinx links for affiliate package convenience.
Outside the astropy package, subclasses of SkyCoord, with inherited docstrings,
will now link to the astropy docs instead of raising a warning / error."

> RST links in SkyCoord now use intersphinx for affiliate package convenience.

Reviewer 2: ```suggestion
Sphinx RST links in ``SkyCoord`` now use intersphinx for downstream packages' convenience.
```

> Sphinx RST links in ``SkyCoord`` now use intersphinx for downstream packages' convenience.

Reviewer: Isn't it all through astropy now? Or is `SkyCoord` still the only thing that could be subclassed and thus be wrong?

Author: yes, it's now all over astropy. I'm moving the note up to the top level. I've contributed a number of such fragments, all about xref, docstrings, and the documentation. Perhaps there should be a specific category / folder for such changelog fragmets, e.g. ``documentation/``.   @saimn how feasible is this with towncrier?

Reviewer 3: It's easy to do but until now the policy has been that we do not mention documentation changes. Exceptions are possible such as for big changes like you did but I'm not sure that justifies a new category. "other" seems fine to me. @pllim @bsipocz - Thoughts ?

Reviewer 4: I don't think this would need a changelog to begin with. And as a rule of thumb, the changelog doesn't have any links. The links tend to broke on the many year timescale, and while SkyCoord is expected not to be one of those, the blanket rule of no links works better than trying to maintain a list of OK and not OK types of links.

> Outside the astropy package, subclasses of SkyCoord, with inherited docstrings,

Reviewer 2: ```suggestion
Outside the ``astropy`` package, subclasses of ``SkyCoord``, with inherited docstrings,
```

> will now link to the astropy docs instead of raising a warning / error.

Reviewer 2: ```suggestion
will now link to the ``astropy`` docs instead of raising a warning or an error.
```
